client/src/components/Game/Timer.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { withRouter } from 'react-router';
+import { compose } from 'recompose';
+
+class Timer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      minutes: 99,
+      seconds: 99,
+      initialSet: false
+    }
+  }
+
+  async componentDidMount() {
+    const { id } = this.props.match.params;
+    const game = this.props.firebase.game(id);
+
+    game.get()
+    .then(docSnapshot => {
+      if(docSnapshot.exists) {
+        game.onSnapshot((snapshot) => {
+          if(snapshot.data().gameStartTime !== null) {
+            this.setState({
+              gameSize: snapshot.data().players.length,
+              startTime: snapshot.data().gameStartTime.seconds,
+              timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+            })
+            this.setState({
+              timeTurnWillEnd: ((this.state.timePerTurnInSeconds * this.state.gameSize) + this.state.startTime),
```

Comment:
I'm a little confused about this calculation. Isn't this the ceiling time at which the entire game will end, and not one turn?

Reply:
that's may be a fault of my bad variable naming, gameSize should be named amountOfPlayers.

Reply:
So if I had 10 players, 10 minutes per player, the first players would have 10*10 = 100 minutes to finish their drawing after game start, before their turn ends?

Reply:
that was bad brain moment lol, it should be called currentPlayerIndex

Reply:
So in the same situation, if the first 4 players take 5 minutes each, does the 5th player get 30 minutes?

Reply:
so lets say the game starts at 3:00 pm with 5 minute turns. 
- first player timeTurnWillEnd: 300 seconds * 1 past 3 pm (3:05)
- second player timeTurnWillEnd: 300 seconds * 2 past 3 pm (3:10)
and so on...i just realized however that my code doesn't reflect this, so i am going to fix it

Reply:
Yes, here is my example.
Say the first player takes 1 minute. Does this mean the 2nd player gets 9 minutes?

Reply:
ohhh okay i understand. i have to add a reset function in case some one actually sends before the time is up